integration-tests/src/integrationTest/org/apache/cassandra/sidecar/health/JMXReconnectTest.java

```diff
@@ -0,0 +1,249 @@
+                    commAdminField.setAccessible(true);
+                    Object obj = commAdminField.get(rmiConnector);
+
+                    // Call RMIClientCommunicatorAdmin.gotIOException
+                    Method m = obj.getClass().getDeclaredMethod("gotIOException", IOException.class);
+                    m.setAccessible(true);
+                    m.invoke(obj, new NoSuchObjectException("injected"));
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new RuntimeException(ex);
+            }
+        }
+    }
+
+    private static File createBootstrapJar() throws Exception
+    {
+        File jar = File.createTempFile("bb-bootstrap", ".jar");
+        jar.deleteOnExit();
+
+        try (JarOutputStream jos = new JarOutputStream(new FileOutputStream(jar)))
+        {
+            addClassToJar(jos, ObjectHolder.class);
+            addClassToJar(jos, ConstructorAdvice.class);
+            addClassToJar(jos, GetAttributeAdvice.class);
+        }
+
+        return jar;
+    }
+
+    private static void addClassToJar(JarOutputStream jos, Class<?> clazz) throws IOException
+    {
+        String name = clazz.getName().replace('.', '/') + ".class";
+        jos.putNextEntry(new JarEntry(name));
+        try (InputStream is = clazz.getClassLoader().getResourceAsStream(name))
+        {
+            Assertions.assertNotNull(is);
+            is.transferTo(jos);
+        }
+        jos.closeEntry();
+    }
+
+    /**
+     * Tests JMX reconnection handling to prevent deadlocks in ClientCommunicatorAdmin.restart.
+     * <p>
+     * This test simulates a specific deadlock scenario that can occur during JMX reconnection:
+     * <ol>
+     *   <li>JMX connection undergoes reconnection process</li>
+     *   <li>A notification handler for connection status changes executes JMX calls during reconnection</li>
+     *   <li>The JMX call fails with an IOException</li>
+     *   <li>This sequence creates a potential deadlock in ClientCommunicatorAdmin.restart</li>
+     * </ol>
+     * <p>
+     * The test uses bytecode instrumentation to inject IOException failures at specific points in the
+     * JMX reconnection flow, then validates that the system can recover and continue processing
+     * JMX requests successfully.
+     *
+     * @throws ClassNotFoundException    if reflection fails to find required classes
+     * @throws NoSuchMethodException     if reflection fails to find required methods
+     * @throws InvocationTargetException if method invocation fails during reflection
+     * @throws IllegalAccessException    if reflection access is denied
+     * @throws NoSuchFieldException      if reflection fails to find required fields
+     */
+    @Test
+    void testJMXReconnect() throws ClassNotFoundException, NoSuchMethodException, InvocationTargetException, IllegalAccessException, NoSuchFieldException
```

Comment:
Add Timeout for this test maybe?
```suggestion
    @Test
    void testJMXReconnect() throws Exception
```

Reply:
done